source/val/validate_mesh_shading.cpp

```diff
@@ -103,15 +122,48 @@ spv_result_t MeshShadingPass(ValidationState_t& _, const Instruction* inst) {
         return _.diag(SPV_ERROR_INVALID_DATA, inst)
                << "Primitive Count must be a 32-bit unsigned int scalar";
       }
-
+      
       break;
     }
 
     case spv::Op::OpWritePackedPrimitiveIndices4x8NV: {
       // No validation rules (for the moment).
       break;
     }
-
+    case spv::Op::OpVariable: {
+      if (_.HasCapability(spv::Capability::MeshShadingEXT)) {
+        bool meshInterfaceVar = IsInterfaceVariable(
+            _, inst, spv::ExecutionModel::MeshEXT);
+        bool fragInterfaceVar = IsInterfaceVariable(
+            _, inst, spv::ExecutionModel::Fragment);
+
+        const spv::StorageClass storage_class =
+            inst->GetOperandAs<spv::StorageClass>(2);
+        bool storage_output = (storage_class == spv::StorageClass::Output);
+        bool storage_input = (storage_class == spv::StorageClass::Input);
+        
+        if (_.HasDecoration(inst->id(), spv::Decoration::PerPrimitiveEXT)) {
+          if (fragInterfaceVar && !storage_input) {
+            return _.diag(SPV_ERROR_INVALID_DATA, inst)
+                   << "PerPrimitiveEXT decoration must be applied only to "
+                      "variables in the Input Storage Class in the Fragment "
+                      "Execution Model.";
+          }
+
+          if (meshInterfaceVar && !storage_output) {
+            std::string vkerror = (spvIsVulkanEnv(_.context()->target_env))
```

Comment:
`VkErrorID` already will return `""` if its not a Vulkan Env, so you don't need to do that here

Reply:
I intend to print the error irrespective of whether the SPIRV is used for vulkan environment or not. That is the reason.

Reply:
yes, if you go just go `_.VkErrorID(4336)` it will already return the correct message for you regardless of the environment